### PR TITLE
Support for Solidity 0.8.24

### PR DESCRIPTION
**Pull request by blitz-1306**
## Changes
- [x] Update dependencies.
- [x] Add support for Solidity 0.8.24. On compiler-level (there are no related changes in AST).
- [x] Support newly introduced builtins in type inference components.
- [x] Add few missing tests to improve coverage.

> [!NOTE]
> Introduced support for `cancun` is not yet **default**. To compile related samples, add following CLI option:
> ```
> --compiler-settings '{"evmVersion": "cancun"}'
> ```

## Related links
- [Solidity 0.8.24 release notes](https://github.com/ethereum/solidity/releases/tag/v0.8.24)
- [Solidity 0.8.24 release blog post](https://soliditylang.org/blog/2024/01/26/transient-storage/)
- [Diff of Solidity 0.8.24 and Solidity 0.8.23](https://github.com/ethereum/solidity/compare/v0.8.23...v0.8.24)

Regards.

### CODE REVIEW

File: test/samples/solidity/latest_08.sol [hunk 1]
```diff
@@ -413,3 +413,25 @@ contract Features_0822 {
         emit Y(5);
     }
 }
+
```
Comment (cd1m0) [inline test/samples/solidity/latest_08.sol:416]: Can we separate this as a separate sample with proper args? Everything else looks great

**Review by cd1m0 — CHANGES_REQUESTED** [review]
Looks good! One nit

File: test/samples/solidity/latest_08.sol (hunk 1 above)
Comment (blitz-1306) [reply test/samples/solidity/latest_08.sol]: Addressed in 0d19096

**Review by cd1m0 — APPROVED** [review]
lgtm

**Pull request merged**